### PR TITLE
fix: fast ssr access to globals should be allowed ouside of a request after adding RequestStorageManager middleware

### PR DESCRIPTION
**Pull request by erhuan-msft**
# Pull Request

## 📖 Description

During integration of fast-ssr `RequestStorageManager` middleware to a SSR server, we discovered access to several globals will throw exceptions if code is not running under storage scope (ie. inside a request). This change will ensure the original global values are assessible outside of a request.

This issue was discovered when running unit test on the test server. Jest will try to access and reset the globals within its framework and thereby causing the exceptions to occur.

Thanks to @EisenbergEffect for the discussion and the code snippet.

## 📑 Test Plan

Unit tests added for change in `installDOMShim` behavior

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "After install DOM Shim, globals will now return the original global instead of the shimmed version in async local storage if code is not in storage scope.",
+  "packageName": "@microsoft/fast-ssr",
+  "email": "erhuan@microsoft.com",
+  "dependentChangeType": "patch"
```
Comment (chrisdholt) [inline change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json:6]: This should also be prerelease

Comment (erhuan-msft) [reply change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json:6]: I did select prerelease under the `yarn change` prompt. Can I just hand edit this file?

Comment (EisenbergEffect) [reply change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json:6]: Yes, you can just hand edit that and commit. It's a bit of a quick of beachball for prerelease packages.

Comment (EisenbergEffect) [inline change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

**Review by EisenbergEffect — APPROVED** [review]
Approved pending the small fix to the change file. Thanks for adding the tests!

**Review by nicholasrice — APPROVED** [review]
Thanks @erhuan-msft!

File: change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json (hunk 1 above)
Comment (erhuan-msft) [reply change/@microsoft-fast-ssr-e107db57-9534-4ac4-bd65-157d5b40a79f.json]: Resolved now

**Pull request merged**